components/ParameterForm.vue

```diff
@@ -120,14 +173,117 @@ const renderAsSelect = (param: Parameter) => {
   return !renderAsRadios(param) && [TypeOfParameter.Select, TypeOfParameter.GlobeSelect].includes(param.parameterType);
 };
 
+// Retrieve the ValueData for a (numeric) parameter that is dependent on the value of another parameter.
+const numericUpdateData = (param: Parameter): ValueData | undefined => {
+  if (!param.updateNumericFrom || !formData.value) {
+    return;
+  }
+
+  const foreignParamId = param.updateNumericFrom.parameterId;
+  const foreignParamVal = formData.value[foreignParamId];
+  if (param.updateNumericFrom && typeof foreignParamVal !== "undefined") {
+    return param.updateNumericFrom?.values[foreignParamVal.toString()];
+  }
+};
+
+const min = (param: Parameter) => {
+  return numericUpdateDict.value[param.id]?.min;
+};
+
+const max = (param: Parameter) => {
+  return numericUpdateDict.value[param.id]?.max;
+};
+
+const invalidFields = computed(() => {
+  if (!formData.value && paramMetadata.value) {
+    return paramMetadata.value.map(param => param.id);
+  }
+
+  const invalids = Array<string>();
+  paramMetadata.value?.forEach((param) => {
+    if (formData.value![param.id] === "") {
+      invalids.push(param.id);
+    };
+
+    if (param.parameterType === TypeOfParameter.Numeric && param.updateNumericFrom) {
+      const val = Number.parseInt(formData.value![param.id] as string);
+      const { min, max } = numericUpdateData(param) as ValueData;
+
+      if (val < min || val > max) {
+        invalids.push(param.id);
+      }
+    };
+  });
+
+  return invalids;
+});
+
+const defaultValue = (param: Parameter) => {
+  if (param.updateNumericFrom) {
+    const foreignParamId = param.updateNumericFrom.parameterId;
+    return param.updateNumericFrom.values[formData.value![foreignParamId]].default.toString();
+  } else if (param.parameterType === TypeOfParameter.Select || param.parameterType === TypeOfParameter.GlobeSelect) {
+    return param.defaultOption || param.options[0].id;
+  }
+  // Currently, due to the metadata schema, non-updatable numerics don't have default values available.
+};
+
+const valueIsAtDefault = (param: Parameter) => {
+  return formData.value![param.id] === defaultValue(param);
+};
+
+const resetParam = (param: Parameter) => {
+  formData.value![param.id] = defaultValue(param) as string;
+};
+
+const numericParameterFeedback = (param: Parameter) => {
+  if (param.updateNumericFrom) {
+    const foreignParamId = param.updateNumericFrom.parameterId;
+    const foreignParamOptionLabel = appStore.metadata!.parameters.find(param => param.id === foreignParamId)!
+      .options.find(option => option.id === formData.value![foreignParamId])?.label;
+    return `${min(param)} to ${max(param)} is the allowed ${param.label.toLowerCase()} range for ${foreignParamOptionLabel}.`;
+  }
+};
+
+const pulse = (parameterId: string) => {
+  const input = document.getElementById(parameterId);
```

Comment:
Ah, that's a much nicer way of doing it. I was playing around with using 'refs' (as attributes in the markup) to refer to the elements to act on -- but since they had to have dynamic ids (from metadata) that proved too difficult.